tasks/Resources/SERH_OLM/install.resource

```diff
@@ -0,0 +1,145 @@
+    Run Keyword And Continue On Failure    Verify Deployment  component=${prometheus}      nPods=1  nContainers=2  containerNames=${prom_containerNames}
+    @{prometheus_operator}=  Oc Get    kind=Pod  namespace=${STARBURST_CR_DEFAULT_NAMESPACE}  label_selector=k8s-app=prometheus-operator
+    ${prom_op_containerNames}=  Create List  prometheus-operator
+    Run Keyword And Continue On Failure    Verify Deployment  component=${prometheus_operator}      nPods=1  nContainers=1  containerNames=${prom_op_containerNames}
+
+Managed Starburst Pods Should Be Deployed
+    [Documentation]    Verifies the Starburst has the expected pods and containers
+    ${running_pods}=     Run    oc get pods -n ${STARBURST_CR_DEFAULT_NAMESPACE} -o json | jq '.items[] | select(.status.phase=="Running") | .metadata.name' | sed 's/"//g'
+    ${completed_pods}=     Run    oc get pods -n ${STARBURST_CR_DEFAULT_NAMESPACE} -o json | jq '.items[] | select(.status.phase=="Succeeded") | .metadata.name' | sed 's/"//g'
+    ${running_pods_list}=    Text To List    ${running_pods}
+    ${completed_pods_list}=    Text To List    ${completed_pods}
+    Run Keyword And Continue On Failure    Length Should Be    ${running_pods_list}    7
+    Run Keyword And Continue On Failure    Length Should Be    ${completed_pods_list}    2
+    Verify Coordinator And Workers Deployment
+    Verify Prometheus Deployment
+  
+Wait Until Managed Starburst Pods Are Deployed
+    [Documentation]    Keeps checking for status of Starburst pods until
+    ...                they gets deployed
+    [Arguments]    ${retries}=10   ${retries_interval}=30s
+    FOR  ${retry_idx}  IN RANGE  0  1+${retries}
+        Log    Checking Managed Starburst pods are successfully deployed: retry ${retry_idx}    console=yes
+        ${current_status}=   Run Keyword And Return Status    Managed Starburst Pods Should Be Deployed
+        Log    Last status of Managed Starburst CR installation: ${current_status}    console=yes
+        Exit For Loop If    $current_status == True
+        Sleep    ${retries_interval}
+    END
+    IF    $current_status == False
+        Fail    Managed Starburst Pods didn't get deployed after ${retries} retries.
+        ...     (time between retries: ${retries_interval}). Check the cluster...
+    END
+
+Wait Until Managed Starburst Installation Is Completed
+    [Documentation]    Keeps checking for Starburst resources until
+    ...                all of them get the expected status
+    [Arguments]    ${cr_chk_retries}=30   ${cr_chk_retries_interval}=30s
+    ...            ${pods_chk_retries}=30   ${pods_chk_retries_interval}=30s
+    Wait Until Managed Starburst CR Is Deployed      retries=${cr_chk_retries}      retries_interval=${cr_chk_retries_interval}
+    Wait Until Managed Starburst Pods Are Deployed    retries=${pods_chk_retries}    retries_interval=${pods_chk_retries_interval}
+
+Delete Managed Starburst CRs
```

Comment:
Why are we manually deleting resources during testing?

Reply:
because per design, the Addon won't delete the CR

Reply:
so in order to test the uninstallation, we need to remove the CR in the test

Reply:
Why won't the addon delete its own resources during uninstallation?
If the CR being left in the cluster is an expected behaviour after addon uninstallation, then we shouldn't need to delete it ourselves, right? We should actually expect to find it in the cluster after the addon has been uninstalled?

Reply:
we discussed about it in a meeting  about this, and engineers said it's not ideal to delete the Custom Resource which can be managed by the customer. However, it may change in future release, I will raise a bug/story to continue the discussion on this